app/routes/_protected.demo.tsx

```
@@ -0,0 +1,83 @@
+import type {
+  QueryData,
+  RealtimePostgresChangesPayload,
+} from '@supabase/supabase-js';
+import { useEffect, useState } from 'react';
+import { boardwalkDb } from '~/features/boardwalk-db/database';
+import { useUserStore } from '~/features/user/user-provider';
+
+const accountsQuery = boardwalkDb.from('accounts').select();
+type Accounts = QueryData<typeof accountsQuery>;
+type Account = Accounts[number];
+
+export default function Demo() {
+  const user = useUserStore((x) => x.user);
+  const [accounts, setAccounts] = useState<Accounts>([]);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    const getAccounts = async () => {
+      if (!user.id) return;
+
+      const { data, error } = await accountsQuery;
+
+      if (error) {
+        setError(error.message);
+      } else {
+        setAccounts(data);
+      }
+    };
+
+    getAccounts();
+  }, [user.id]);
+
+  useEffect(() => {
+    const channel = boardwalkDb
+      .channel('accounts')
+      .on(
```

Note: there was a silent error because I hadn't added the realtime migration yet. It went away after running `bun supabase migration up`. There were web socket messages with the error in it. Just FYI for the real implementation of this.

---

yeah I know. we can see about error handling when we have actual use case. this is just a demo page